code/datums/components/deployable.dm

```diff
@@ -0,0 +1,62 @@
+/datum/component/deployable
+	var/deploy_time = 5 SECONDS //Default time it takes to deploy the item
+	var/obj/thing_to_be_deployed //What do we spawn when the item is deployed successfully?
+	var/deployed_name //For getting the name of an object for examines later on
+	var/delete_on_use = TRUE //Do we delete the item being used when the object is deployed
+
+/datum/component/deployable/Initialize(deploy_time, thing_to_be_deployed, delete_on_use)
+	. = ..()
+	if(!isitem(parent))
+		return COMPONENT_INCOMPATIBLE
+
+	src.deploy_time = deploy_time
+	src.thing_to_be_deployed = thing_to_be_deployed
+	src.delete_on_use = delete_on_use
+
+	RegisterSignal(parent, COMSIG_PARENT_EXAMINE, .proc/examine)
+	RegisterSignal(parent, COMSIG_ITEM_ATTACK_SELF, .proc/deploy)
```

Comment:
This isn't deploy_signal_handler.

Reply:
```suggestion
	RegisterSignal(parent, COMSIG_ITEM_ATTACK_SELF, .proc/deploy_signal_handler)
```

...though honestly--rename `deploy_signal_handler` to `on_attack_hand`, since it's more accurate.